src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,100 @@
-import { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [isDisabled, setIsDisabled] = useState(true);
```

Comment:
Instead of repeating each state for each input you could do like so:
```suggestion
const [count, setCount] = useState(0);
const [isAddButtonDisabled, setIsAddButtonDisabled] = useState(true);
const [formData, setFormData] = useState({
    title: '',
    description: '',
    imgUrl: '',
    imdbUrl: '',
    imdbId: '',
  });
```